### ISSUE TITLE
PreferenceModelsRegistry does not work for ProxyModel

### ISSUE THREAD

**Issue by danie1k** [EliotBerriot/django-dynamic-preferences#134]
### Background
I have proxy model for `dynamic_preferences.models.GlobalPreferenceModel`:
```python
# my_app/models.py
from dynamic_preferences.models import GlobalPreferenceModel as BaseGlobalPreferenceModel

class GlobalPreferenceModel(BaseGlobalPreferenceModel):
    class Meta:
        proxy = True
        verbose_name = __('Global Preference')
        verbose_name_plural = __('Global Preferences')
```
And custom admin for that model:
```python
# my_app/admin.py
from dynamic_preferences.admin import GlobalPreferenceAdmin as BaseGlobalPreferenceAdmin
from dynamic_preferences.models import GlobalPreferenceModel as BaseGlobalPreferenceModel
from .models import GlobalPreferenceModel

class GlobalPreferenceAdmin(BaseGlobalPreferenceAdmin):
   pass

admin.site.unregister(BaseGlobalPreferenceModel)
admin.site.register(GlobalPreferenceModel, GlobalPreferenceAdmin)

```

### An issue
Because in `DynamicPreferencesConfig.ready` there is native `GlobalPreferenceModel` added to registry, then when I try to save settings on my proxy model & custom admin, I get server error, because  method
```python
# registries.py
class PreferenceModelsRegistry(persisting_theory.Registry):
    ...
    def get_by_preference(self, preference):
        return self[preference.__class__]
```
tries to get key `<class 'my_app.models.GlobalPreferenceModel'>`, but only one key registered in `self` is `<class 'dynamic_preferences.models.GlobalPreferenceModel'>`.

I can register `my_app.models.GlobalPreferenceModel` in `MyAppConfig.ready`, but I'm not sure about data consistency, when two models (native + proxy) will be registered, when in fact they're the same one model.
I'm also not sure what  solution would be the best - e.g. just register proxy model, or do some changes to `PreferenceModelsRegistry`, or even something else.

**Comment by danie1k** [comment]
Looks like some quick fix may help:
```python
    def get_by_preference(self, preference):
        if preference._meta.proxy:
            preference_class = preference._meta.proxy_for_model
        else:
            preference_class = preference.__class__
        return self[preference_class]
```

But I didn't test it widely yet.

**Comment by EliotBerriot** [comment]
Since your issue is about replacing the default model, can you try doing this in your app config?

```
from django.apps import AppConfig

from dynamic_preferences.registries import preference_models, global_preferences_registry


class YourAppConfig(AppConfig):
    name = 'yourapp'
    def ready(self):
        from dynamic_preferences.models import GlobalPreferenceModel
        del preference_models[GlobalPreferenceModel]
        preferences_models.register(YourProxyModel, global_preferences_registry)
```

Basically, it removes the existing mapping between the built-in global preferences model and registry, and replace it with a mapping between your own proxy model and the global preferences registry.

It should work for your use case, but please test this locally before putting it in production. I'll think about some kind of setting to replace the built in models by your own.

**Comment by EliotBerriot** [comment]
Also, it seems GlobalPreferenceModel is referenced directly [in various places](https://github.com/EliotBerriot/django-dynamic-preferences/search?utf8=%E2%9C%93&q=globalpreferencemodel&type=). I'm not sure how my suggestion will work :/

**Comment by danie1k** [comment]
I prefer not to interfere in internal data of third party modules, because it may cause unforeseen consequences on stability or data consistence. That's why I proposed an update to `get_by_preference()` method.

**Comment by EliotBerriot** [comment]
Fixed in #137
